app/javascript/components/mentoring/queue/TrackFilterList.tsx

```diff
@@ -55,39 +60,88 @@ type Props = {
   isFetching: boolean
   value: MentoredTrack | null
   setValue: (value: MentoredTrack) => void
+  cacheKey: string
+  links: {
+    tracks: string
+    updateTracks: string
+  }
 }
 
 const Component = ({
   tracks,
   isFetching,
   value,
   setValue,
-}: Props): JSX.Element => {
-  const handleChange = useCallback(
-    (e, optionValue) => {
-      setValue(optionValue)
+  cacheKey,
+  links,
+}: Props): JSX.Element | null => {
+  const changeTracksRef = useRef<HTMLButtonElement>(null)
+  const handleItemSelect = useCallback(
+    (index) => {
+      if (!tracks) {
+        return
+      }
+
+      const track = tracks[index]
+
+      track ? setValue(tracks[index]) : changeTracksRef.current?.click()
     },
-    [setValue]
+    [setValue, tracks]
   )
+  const {
+    buttonAttributes,
+    panelAttributes,
+    listAttributes,
+    itemAttributes,
+  } = useDropdown((tracks?.length || 0) + 1, handleItemSelect)
+
+  if (!tracks) {
+    return null
+  }
+
+  const selected = tracks.find((track) => track === value) || tracks[0]
 
   return (
-    <div className="track-filter">
-      {isFetching ? <span>Fetching</span> : null}
-      <h3>Filter by language track</h3>
-      {tracks && tracks.length > 0 ? (
-        <div className="tracks">
-          {tracks.map((track) => (
-            <TrackFilter
-              key={track.id}
-              onChange={(e) => handleChange(e, track)}
-              checked={value?.id === track.id}
-              {...track}
-            />
-          ))}
+    <React.Fragment>
+      {isFetching ? <span>Fetching...</span> : null}
+      <button
+        aria-label="Button to open the track filter"
```

Comment:
Is the "Button to" prefix needed? It looks like a screenreader could recognize that it is a button from the `<button>` element itself. But I don't know anything about accessibility to be honest so 🤷

Reply:
Agreed :) I will remove it.